src/vm/win32threadpool.h

```diff
@@ -1132,8 +1141,9 @@ class ThreadpoolMgr
         if (CLRThreadpoolHosted())
             return false;
 
-        DWORD requiredInterval = NextCompletedWorkRequestsTime - PriorCompletedWorkRequestsTime;
-        DWORD elapsedInterval = GetTickCount() - PriorCompletedWorkRequestsTime;
+        DWORD priorTime = PriorCompletedWorkRequestsTime; 
```

Comment:
The volatile load/store implementation is intended for different purposes, for example:

```
done = 0
result = ?

Thread A:
    // do work
    result = ...
    VolatileStore(&done, 1)

Thread B:
while(true) {
    if(VolatileLoad(&done) == 0)
        continue
    ... = result
    break
}
```

With volatile load/store to 'done' as above, it guarantees that when thread B sees 'done == 1', that the value it reads for 'result' will be the latest value written by thread A before setting 'done = 1'. For this usage, the memory barrier for a volatile store needs to be before the store (between writing results and setting completion state), and the volatile load needs to have a memory barrier after the load (between reading completion state and loading results). On x86/x64, the processor already makes this guarantee, so these barriers are not necessary (still need to prevent compiler reordering though).

Unfortunately, volatile gets overused for purposes it's not intended. Due to the memory barrier after the load, a volatile load may read old data. When done in a loop with a barrier, as in thread B, it doesn't make much difference, but on x86/x64 if there is no memory barrier, thread B may never see 'done == 1'.

When periodically reading shared data with only the intent of reading the latest data, the memory barrier should be before the load and after the store (inverse of volatile). These barriers are necessary on x86/x64 as well. This seems to be the scenario here.

In this particular case, we can do a little better with the time variables because the check in ShouldAdjustMaxWorkersActive() compares (Next - Prior) with (currentTime - Prior), so it doesn't matter if we read an old value for Prior, the computation will still produce the right result. Reading the latest Next would help this check be representative of recent state. So I suggested this:

```
Stores        | Loads
-----------------------------
Next = ...    | ... = Prior
MemoryBarrier | MemoryBarrier
Prior = ...   | ... = Next
```

With the memory barrier before the load and after the store of Next, it would read a recent value for Next. And it has the added benefit that the value it reads for Prior will always be equal in age or older than the value it reads for Next, so (Next - Prior) won't turn out to be some huge value after overflow, which would always fail the check and not be useful.

That's for PriorCompletedWorkRequestsTime and NextCompletedWorkRequestsTime. For PriorCompletedWorkRequests, I don't think it matters much where it goes relative to the memory barrier, but I would put the memory barrier before the load and after the store.